goal_src/jak2/pc/features/speedruns.gc

```diff
@@ -16,6 +16,58 @@
 (set! (-> *speedrun-info* active-practice-objective start-zone-init-params) (new 'static 'objective-zone-init-params))
 (set! (-> *speedrun-info* active-practice-objective end-zone-init-params) (new 'static 'objective-zone-init-params))
 
+;; dynamically populate continue-point list, for custom category menu
+(define *main-continue-point-list* '())
+
+(defun is-main-level? ((lvl-name symbol))
+  (case lvl-name
+    (('vinroom 'drillmid 'drill 'drillb 'sewer 'sewerb 'sewesc 'sewescb 'tomba 'tombb 'tombc 'tombd 'tombe 'tombboss 
+      'under 'underb 'palcab 'palshaft 'palboss 'palroof 'throne 'palent 'prison 'forexita 'forexitb 'forresca 'forrescb 
+      'fordumpa 'fordumpb 'fordumpc 'fordumpd 'strip 'ruins 'sagehut 'atoll 'mountain 'forest 'forestb 'mincan 
+      'ctygena 'ctygenb 'ctygenc 'ctysluma 'ctyslumb 'ctyslumc 'ctyport 'ctyfarma 'ctyfarmb 'ctyinda 'consite 'consiteb 
+      'ctyindb 'ctymarka 'ctymarkb 'ctypal 'stadium 'stadiumb 'stadiumc 'stadiumd 'skatea 'garage 'onintent 'kiosk 
+      'oracle 'hideout 'hiphog 'gungame 'dig1 'dig3a 'dig3b 'caspad 'castle 'casboss 'village1 'nest 'nestb) 
+      #t)
+    (else
+      #f)))
+
+(defun is-main-continue? ((cont continue-point))
+  (not (logtest? (-> cont flags) (continue-flags record-path demo-end record-sig demo intro title title-movie))))
+
+(defun build-main-continue-point-list ()
+  ;; reset list just to be safe
+  (set! *main-continue-point-list* '())
+  (let ((lvl-ptr *level-load-list*))
+    ;; loop thru levels
+    (while (not (null? lvl-ptr))
+      (let* ((lvl (the-as level-load-info (-> (the-as symbol (car lvl-ptr)) value)))
+             (continue-ptr (-> lvl continues)))
+        (when (is-main-level? (-> lvl name))
+          ;; loop thru level's continues
+          (while (not (null? continue-ptr))
+            (let ((cont (the-as continue-point (car continue-ptr))))
+              (when (is-main-continue? cont)
+                ;; this continue is valid, push to list
+                (cons! *main-continue-point-list* (-> cont name))))
+            (set! continue-ptr (cdr continue-ptr)))))
+      (set! lvl-ptr (cdr lvl-ptr)))))
```

Comment:
Since the `popup-menu-dynamic-submenu` uses iterates their lists by index, I figured it was better to flatten these once, rather than traverse the entire `*level-load-list*` every time we want to lookup continue points by index